index.ts

```diff
@@ -83,6 +83,102 @@ declare namespace dataSource {
   function create(name: string, params: Array<string>): void
 }
 
+function format(fmt: string, args: string[]): string {
+  let out = ''
+  let argIndex = 0
+  for (let i: i32 = 0; i < fmt.length; i++) {
+    if (i < fmt.length - 1 && fmt.charAt(i) == '{' && fmt.charAt(i + 1) == '}') {
+      if (argIndex >= args.length) {
+        throw Error('Too few arguments for format string: ' + fmt)
+      } else {
+        out += args[argIndex++]
+        i++
+      }
+    } else {
+      out += fmt[i]
+    }
+  }
+  return out
+}
+
+/**
+ * Converts a value to a string, if possible. Throws otherwise.
+ */
+export function toString<T>(v: T): string {
+  if (isInteger<T>(v)) {
+    return v.toString()
+  } else if (isFloat<T>(v)) {
+    return v.toString()
+  } else if (v instanceof bool) {
+    return v ? 'true' : 'false'
+  } else {
+    return v.toString()
+  }
+}
```

Comment:
May be for simplicity better use something like this?
```ts
@inline
export function toString<T>(v: T): string {
  return v.toString()
}
```
Btw `Boolean#toString()` should works: https://webassembly.studio/?f=vf6zrpim5a

Reply:
I'm not sure this function is really useful tbh.

Is there a more convenient way to convert `i32`s etc. to strings than `(100).toString()` (since `100.toString()` doesn't work)?

Reply:
AS follow JS and TS spec in this case. In JS you should use one of two ways:
`(100).toString()`or `100..toString()`. This is all because in JS all numbers are float point by default and allow shorthand notation like `100.`, `1.`, so first dot in number literal always interpret as decimal point)

Reply:
@MaxGraey `v.toString()` may work if `v` is a `bool` but these don't work:
```typescript
let str3 = true.toString()
let str4 = (true).toString()
```

See https://webassembly.studio/?f=9ccvfz53bw7